packages/devtools_app/integration_test/test/live_connection/network_screen_test.dart

```diff
@@ -0,0 +1,109 @@
+
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/shared/table/table.dart' show DevToolsTable;
+import 'package:devtools_test/helpers.dart';
+import 'package:devtools_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart' as http;
+import 'package:integration_test/integration_test.dart';
+
+// To run:
+// dart run integration_test/run_tests.dart --target=integration_test/test/live_connection/network_screen_test.dart --test-app-device=cli
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  late TestApp testApp;
+
+  setUpAll(() {
+    testApp = TestApp.fromEnvironment();
+    expect(testApp.vmServiceUri, isNotNull);
+  });
+
+  tearDown(() async {
+    await resetHistory();
+  });
+
+  testWidgets('nnn', (tester) async {
+    await pumpAndConnectDevTools(tester, testApp);
+    await _prepareNetworkScreen(tester);
+
+    final networkScreen = _NetworkScreenHelper(tester, testApp.controlPort!);
+
+    // Instruct the app to make a GET request via the dart:io HttpClient.
+    await networkScreen.triggerRequest('get/');
+    _expectInRequestTable('GET');
+    await networkScreen.clear();
+
+    // Instruct the app to make a POST request via the dart:io HttpClient.
+    await networkScreen.triggerRequest('post/');
+    _expectInRequestTable('POST');
+    await networkScreen.clear();
+
+    // Instruct the app to make a PUT request via the dart:io HttpClient.
+    await networkScreen.triggerRequest('put/');
+    _expectInRequestTable('PUT');
+    await networkScreen.clear();
+
+    // Instruct the app to make a DELETE request via the dart:io HttpClient.
+    await networkScreen.triggerRequest('delete/');
+    _expectInRequestTable('DELETE');
+    await networkScreen.clear();
+
+    // Instruct the app to make a GET request via Dio.
+    await networkScreen.triggerRequest('dio/get/');
+    _expectInRequestTable('GET');
+    await networkScreen.clear();
+
+    // Instruct the app to make a POST request via Dio.
+    await networkScreen.triggerRequest('dio/post/');
+    _expectInRequestTable('POST');
+
+    await disconnectFromTestApp(tester);
+  });
+}
+
+final class _NetworkScreenHelper {
+  _NetworkScreenHelper(this._tester, this._controlPort);
+
+  final WidgetTester _tester;
+
+  final int _controlPort;
+
+  Future<void> clear() async {
+    // Press the 'Clear' button between tests.
+    await _tester.tap(find.text('Clear'));
+    await _tester.pump(safePumpDuration);
+  }
+
+  Future<void> triggerRequest(String path) async {
+    await http.get(Uri.parse('http://localhost:$_controlPort/$path'));
+    await Future.delayed(const Duration(milliseconds: 200));
+    await _tester.pump(safePumpDuration);
+  }
+}
+
+final _requestTable = find.byType(DevToolsTable<NetworkRequest>);
+
+void _expectInRequestTable(String text) {
+  expect(
+    find.descendant(of: _requestTable, matching: find.text(text)),
```

Comment:
nit: consider defining the finder here directly since the `_requestTable` variable is only used once

Reply:
Done